src/components/login/index.tsx

```diff
@@ -1,50 +1,93 @@
 'use client';
 
-import { useRef } from 'react';
+import { useRef, useState } from 'react';
 import Button from '@mui/material/Button';
 import TextField from '@mui/material/TextField';
 import logo from '@/assets/images/sumtimeLogo.png';
 import { signIn } from 'next-auth/react';
+import { useRouter } from 'next/navigation';
+import { EMAIL_REG_EXP } from '@/constants';
 import * as S from './Login.styled';
 
 function LoginSection() {
+  const router = useRouter();
+
   const emailInputRef = useRef<HTMLInputElement>(null);
   const passwordInputRef = useRef<HTMLInputElement>(null);
 
+  const [emailError, setEmailError] = useState<string | null>(null);
+  const [passwordError, setPasswordError] = useState<string | null>(null);
+
+  let checkLoginForm = false;
+
+  const handleEmailValidation = () => {
+    if (!emailInputRef.current?.value || !EMAIL_REG_EXP.test(emailInputRef.current.value)) {
+      setEmailError('유효한 이메일 주소를 입력해주세요');
+      checkLoginForm = false;
+    } else {
+      setEmailError(null);
+      checkLoginForm = true;
+    }
+  };
+
+  const handlePasswordValidation = () => {
+    if (!passwordInputRef.current?.value) {
+      setPasswordError('비밀번호를 입력해주세요');
+      checkLoginForm = false;
+    } else {
+      setPasswordError(null);
+      checkLoginForm = true;
+    }
+  };
+
   const handleSignIn = async () => {
-    await signIn('credentials', {
-      email: emailInputRef.current?.value,
-      password: passwordInputRef.current?.value,
-      redirect: true,
-      callbackUrl: '/todo',
-    });
+    handleEmailValidation(); // 추가: 이메일 검증
+    handlePasswordValidation(); // 추가: 비밀번호 검증
+
+    if (checkLoginForm) {
```

Comment:
p4:

```suggestion
    if (!emailError && !passwordError) {
```
let으로 변수를 선언할수도 있지만 이렇게 쓸 수도 있을 것 같아요~

Reply:
그렇네요! 변수 선언 하지 않고 저렇게 작성하면 코드가 더 간결해질 것 같습니다~!! 반영 후 커밋하겠습니다!